File: 프로그래머스/1/86491. 최소직사각형/최소직사각형.py
def solution(sizes):
    max_W, max_H = 0, 0
    for size in sizes:
        W, H = size
        a_size = max(max_W, W) * max(max_H, H)
        b_size = max(max_W, H) * max(max_H, W)
        if a_size > b_size:
            max_W = max(max_W, H); max_H = max(max_H, W)
        else:
            max_W = max(max_W, W); max_H = max(max_H, H)
    return max_W * max_H

File: 프로그래머스/1/86491. 최소직사각형/최소직사각형.cpp
#include <string>
#include <vector>
#include <cmath>

using namespace std;

int solution(vector<vector<int>> sizes) {
    
    int W = sizes[0][0]; int H = sizes[0][1];
    for (int i = 1; i < sizes.size(); ++i)
    {
        vector<int> size = sizes[i];
        if ((abs(W - size[0]) + abs(H - size[1])) < (abs(W - size[1]) + abs(H - size[0])))
        {
            W = max(W, size[0]); H = max(H, size[1]);
        }
        else
        {
            W = max(W, size[1]); H = max(H, size[0]);
        }
    }
    
    return W * H;
}